src/Models/ExtensionWrapper.cs

```diff
@@ -182,6 +183,24 @@ public void SignalDispose()
         return GetExtensionObject()?.GetProvider(_providerTypeMap[typeof(T)]) as T;
     }
 
+    public async Task<IEnumerable<T>> GetListOfProvidersAsync<T>()

```

Comment:
What is this for?

Reply:
We wanted to allow an extension to support multiple provider interfaces. E.g an extension can say "Hey i support multiple compute system providers"